extensions/bitwarden/src/create-login.tsx

```diff
@@ -0,0 +1,187 @@
+      return next;
+    });
+  };
+
+  useEffect(() => {
+    focus("name");
+  }, []);
+
+  const generatePassword = async () => {
+    if (isSubmitting) return;
+
+    const toast = await showToast({ title: "Generating password...", style: Toast.Style.Animated });
+
+    try {
+      const options = await getPasswordGeneratorOptions();
+      const generatedPassword = await bitwarden.generatePassword(options);
+      itemProps.visiblePassword.onChange?.(generatedPassword);
+      itemProps.hiddenPassword.onChange?.(generatedPassword);
+      await Clipboard.copy(generatedPassword);
+      toast.title = "Password generated and copied";
+      toast.style = Toast.Style.Success;
+    } catch (error) {
+      toast.title = "Failed to generate password";
+      toast.style = Toast.Style.Failure;
+    }
+  };
+
+  return (
+    <Form
+      isLoading={isSubmitting}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Create Login" onSubmit={handleSubmit} icon={Icon.NewDocument} />
+          <Action
+            icon={showPassword ? Icon.EyeDisabled : Icon.Eye}
+            title={showPassword ? "Hide Password" : "Show Password"}
+            onAction={togglePasswordVisibility}
+            shortcut={{ modifiers: ["cmd"], key: "e" }}
+          />
+          <Action
+            icon={Icon.Key}
+            title="Generate Password"
+            onAction={generatePassword}
+            shortcut={{ modifiers: ["cmd"], key: "g" }}
+          />
+          <DebuggingBugReportingActionSection />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField {...itemProps.name} title="Name" placeholder="GitHub, Gmail" storeValue={false} />
+      <Form.Dropdown {...itemProps.folderId} title="Folder" placeholder="Select folder" storeValue={false}>
+        {folders.map((folder) => (
+          <Form.Dropdown.Item
+            key={folder.id}
+            value={folder.id ?? FOLDER_OPTIONS.NO_FOLDER}
+            title={folder.name}
+            icon={Icon.Folder}
+          />
+        ))}
+      </Form.Dropdown>
+      <Form.TextField {...itemProps.username} title="Username" placeholder="john.doe@mail.com" storeValue={false} />
+      <Form.TextField {...itemProps.uri} title="Website URI" placeholder="example.com" storeValue={false} />
+      {showPassword ? (
+        <Form.TextField
+          {...itemProps.visiblePassword}
+          title="Password"
+          placeholder="Enter password"
+          storeValue={false}
+          onChange={(value) => {
+            itemProps.visiblePassword.onChange?.(value);
+            itemProps.hiddenPassword.onChange?.(value);
+          }}
+        />
+      ) : (
+        <Form.PasswordField
+          {...itemProps.hiddenPassword}
+          onChange={(value) => {
+            itemProps.hiddenPassword.onChange?.(value);
+            itemProps.visiblePassword.onChange?.(value);
+          }}
+          title="Password"
+          placeholder="Enter password"
+          storeValue={false}
```

Comment:
Can you define something like:
```ts
const passwordFieldProps: Form.TextField.Props = {
  title: "Password",
  placeholder: "Enter password",
  onChange: (value) => {
    itemProps.hiddenPassword.onChange?.(value);
    itemProps.visiblePassword.onChange?.(value);
  },
};
```

And then you can have the common props be spread like:

```suggestion
          {...itemProps.visiblePassword}
          {...passwordFieldProps}
        />
      ) : (
        <Form.PasswordField
          {...itemProps.hiddenPassword}
          {...passwordFieldProps}
```